src/PublicApiAnalyzers/Core/Analyzers/DeclarePublicApiAnalyzer.Impl.cs

```diff
@@ -703,6 +754,84 @@ private static bool CanTypeBeExtendedPubliclyImpl(ITypeSymbol type)
                         m => m.DeclaredAccessibility != Accessibility.Internal && m.DeclaredAccessibility != Accessibility.Private && m.DeclaredAccessibility != Accessibility.ProtectedAndInternal
                     );
             }
+
+            /// <summary>
+            /// Various Visit* methods return true if an oblivious reference type is detected.
+            /// </summary>
+            private sealed class ObliviousDetector : SymbolVisitor<bool>
+            {
+                private bool _isTypeDeclaration = true;
+
+                public override bool VisitField(IFieldSymbol symbol)
+                {
+                    _isTypeDeclaration = false;
+                    return Visit(symbol.Type);
+                }
+
+                public override bool VisitMethod(IMethodSymbol symbol)
+                {
+                    _isTypeDeclaration = false;
+                    if (Visit(symbol.ReturnType))
+                    {
+                        return true;
+                    }
+
+                    foreach (var parameter in symbol.Parameters)
+                    {
+                        if (Visit(parameter.Type))
+                        {
+                            return true;
+                        }
+                    }
+
+                    return false;
+                }
+
+                public override bool VisitNamedType(INamedTypeSymbol symbol)
```

Comment:
Where do type parameters which have no constraints get handled in this checker? Sorry, feel like I'm missing the obvious.

Reply:
I didn't follow. It's okay to have unconstrained type parameters in your API (we shouldn't warn about that).

Reply:
What if the unconsrtained type parameter is in an oblivious context

Reply:
Basically this case:

```cs
#nullable enable
class ReallyMean<
#nullable disable
T> { }
```

Missing where we catch this case (or possibly it just doesnt' matter)

Reply:
We can't put annotations in that position `class ReallyMean<T$$>`, be it `?`, `!` or `~` (in spirit).